tests/breed-hpprocurveN1178Test.php

```diff
@@ -0,0 +1,143 @@
+<?php
+
+require_once '../wwwroot/inc/deviceconfig.php';
+
+class hpprocurveN1178Test extends RTTestCase
+{
+
+	// hpprocurveN1178Read8021QConfig
+	/**
+	 * @dataProvider providerHP8021QConfigSamples
+	 */
+
+	public function testHP8021QConfParser ($test_data)
+	{
+	foreach ($test_data as $test => $test_array)
```

Comment:
There is an implied `foreach` between the provider and the test function, just make sure the arrays are nested properly in the provider function and then the test function can be written to handle exactly one case, PHPUnit will take care of driving the process. See `PureFunctionTest.php` for examples. It looks like you could just add items to `providerBinaryEquals()` and avoid a custom test class at all.

Reply:
A possible way to keep it readable in `PureFunctionTest.php` would be to have a new helper method like `providerHP8021Q()` and then to move the existing tests to `providerOtherBinaryFunctions()` and have `providerBinaryEquals()` return the full set with something like `array_merge ($this->providerOtherBinaryFunctions(), $this->providerHP8021Q())`